src/test/java/com/passbook/sparkeighteen/service/TransactionServiceTest.java

```diff
@@ -0,0 +1,220 @@
+
+    @Test
+    public void InvalidPayload_userIdNotFound_transactUnsuccessfull() throws Exception {
+
+        Integer userID = 1;
+
+        TransactionRequest request = TransactionRequest.builder()
+                .amount(250f)
+                .note("Coldrinks")
+                .name("Ketan")
+                .transactionType(TransactionType.CREDIT)
+                .build();
+
+        Mockito.when(userRepository.findById(userID)).thenReturn(Optional.empty());
+
+        TransactionResponse response = transactionService.transact(userID, request);
+        assertEquals("User not found. Please use a registered user", response.getMessage());
+
+    }
+
+    @Test
+    public void transact_userIdExist_transactSuccessful() throws Exception {
+        Integer userID = 1;
+
+        TransactionRequest request = TransactionRequest.builder()
+                .amount(250f)
+                .note("Coldrinks")
+                .name("Ketan")
+                .transactionType(TransactionType.CREDIT)
+                .build();
+
+        UserEntity user = UserEntity.builder()
+                .email("ketan@gmail.com")
+                .dob(LocalDate.of(2000, 2, 25))
+                .password("qwerty")
+                .gender(Gender.MALE)
+                .lastname("Shinde")
+                .firstname("ketan")
+                .build();
+
+        TransactionEntity transaction = TransactionEntity.builder()
+                .amount(request.getAmount())
+                .id(1)
+                .note(request.getNote())
+                .user(user)
+                .time(LocalDateTime.now())
+                .transactionType(request.getTransactionType())
+                .build();
+
+        Mockito.when(userRepository.findById(userID)).thenReturn(Optional.ofNullable(user));
+        Mockito.when(transactionRepository.save(ArgumentMatchers.any())).thenReturn(transaction);
+
+        TransactionResponse response = transactionService.transact(userID, request);
+        assertEquals(String.format("Your A/C XXXXX has a %s by Rs %.2f on %s ", transaction.getTransactionType(), transaction.getAmount(), transaction.getTime()), response.getMessage());
+
+    }
+
+    @Test
+    public void transact_amountLessThanOne_txnTypeCredit_transactUnsuccessful() throws Exception {
+        Integer userID = 1;
+
+        TransactionRequest request = TransactionRequest.builder()
+                .amount(0f)
+                .note("Coldrinks")
+                .name("Ketan")
+                .transactionType(TransactionType.CREDIT)
+                .build();
+
+        TransactionEntity txn = TransactionEntity.builder()
+                .amount(request.getAmount())
+                .build();
+
+        TransactionResponse response = transactionService.transact(userID, request);
+        assertEquals("Invalid Transaction amount", response.getMessage());
+    }
+
+    @Test
+    public void transact_Credit_TransactionSuccessful() throws Exception {
```

Comment:
naming convention ? (methodname_senario_result)

Reply:
@Ketan-SP18 it would be like makeTransaction_Credit_TransactionSuccessful() correct me if i am wrong.

Reply:
It shows similar understanding